test/test_assign.py

```diff
@@ -63,5 +64,15 @@ def test_post_permuted_assignment(self):
 
   # TODO: is there a way to sneak in a permute such that it returns the wrong answer?
 
+  def test_cast_assignment(self):
+    a = Tensor(np.arange(N*N, dtype=np.float32)).reshape(N,N)
+    a.realize()
+    oba1 = a.lazydata.output_buffer
+    a.assign(a.cast(dtypes.float64).realize())
```

Comment:
float64 doesn't have full platform support, maybe use one of the int dtypes instead?

Reply:
I'll change to int32.